### PR TITLE
Fix #24 Pass lambda to default

### LINKED ISSUE #24: Add support for lazy default
Add way to pass lambda to `default` for lazy initialization of default. like:
```
val myOpt by parser.storing("--foo", help = "bar")
    .default { "Do $crazy stuff here, only if there is no myOpt" }
```

### PR DESCRIPTION
**Pull request by shanethehat**

### CODE REVIEW

File: src/main/kotlin/com/xenomachina/argparser/Default.kt [hunk 1]
```diff
@@ -26,6 +26,15 @@ import com.xenomachina.common.Holder
...
 fun <T> ArgParser.DelegateProvider<T>.default(newDefault: T): ArgParser.DelegateProvider<T> {
+    return ArgParser.DelegateProvider(ctor = ctor, defaultHolder = Holder { newDefault })
+}
+
+/**
+ * Returns a new `DelegateProvider` with the specified default value from a lambda.
+ *
+ * @param newDefault the default value for the resulting [ArgParser.Delegate]
+ */
+fun <T> ArgParser.DelegateProvider<T>.default(newDefault: () -> T): ArgParser.DelegateProvider<T> {
```
Comment (xenomachina) [inline src/main/kotlin/com/xenomachina/argparser/Default.kt]: I am a tiny bit nervous about this causing some sort of ambiguity, but I honestly can't think of a situation where a `T` and a `() -> T` could be confused.

File: src/test/kotlin/com/xenomachina/argparser/ArgParserTest.kt [hunk 2]
```diff
@@ -421,6 +421,24 @@ class DefaultWithProviderTest : Test({
     Args(parserOf("-x9", "-x8")).x shouldBe 8
 })
 
+class DefaultWithLambda : Test({
+
+    var a = 1
```
Comment (xenomachina) [inline src/test/kotlin/com/xenomachina/argparser/ArgParserTest.kt]: This is a bit hard to follow. Why not have `var defaultCalled = false` inside `Args`? `defaultCalled shouldBe false` is more self-documenting than `a shouldBe 1`.

**Comment by shanethehat** [comment]
PR updated. I'm also not sure what ambiguity could arise from the remaining comment.
